src/Charts/chart.ts

```diff
@@ -127,11 +133,11 @@ export abstract class Chart {
   }
 
   get extent(): Record<string, number[] | Date[] | null[]> {
-    if (!this._extent) {
-      this._extent = []
-      for (const key in this.dataKeys) {
-        const path = this.dataKeys[key]
-        this._extent[path] = dataExtentFor(this._data, path)
+    if (!this._extent) this._extent = {}
+    for (const key in this.dataKeys) {
+      const path = this.dataKeys[key]
+      if (path in this._extent || this._extent[path] === undefined) {
```

Comment:
Is this not always true? If path is not in `this._extent`, `this._extent[path]` will always be `undefined`?

Reply:
The second gives an error if path is not a propery of this._extent

Reply:
Is `this._extent` not a regular JavaScript object? Also, in that case it will always error if `path` is not in `this._extent`, since it is `||` and not `&&`?